### DISCUSSION TITLE
`meltano elt` plugins_context on starting event

### DISCUSSION THREAD

**Discussion by pnadolny13** [meltano/meltano#6256] [General]
Related to https://github.com/meltano/meltano/pull/6068

It looks like the `meltano elt` telemetry event does get the plugin context but it attaches it to the final event (i.e. completed), for `meltano invoke` we get it as part of the first event (i.e. started). I remember invoke originally having it as the final event as well but we moved it to the first event so that we would never have a case where we might not get the plugins. I'm thinking of a hard exit or maybe a failed sync but I know we have exit events with plugins context which tries and catch failures now.

@WillDaSilva @pandemicsyn What do we think about moving the plugin_context to the starting event for `meltano elt`? Is that a heavy or light change?

**Comment by pandemicsyn** [comment]
Attaching plugin contexts to start events is actually not going to work anymore when we switch to firing an automatic "start" events (because at that stage we don't know anything about the plugins that will eventually be used). 

So, we'll either need to attach them to final events or create a special event to fire when a PluginsContext is found.

**Reply by pandemicsyn** [reply]
Actually `run` doesn't attach them to start either. I think we just need to make sure that we attach the plugins context to the tracker object as soon as we're able to. Then we'd always receive it attached to exit event, unless the failure happens before we can create it - in which case we're SOL anyway?
